chain/checkpointing/network.go

```diff
@@ -0,0 +1,61 @@
+package checkpointing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Zondax/multi-party-sig/pkg/party"
+	"github.com/Zondax/multi-party-sig/pkg/protocol"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+type Network struct {
+	sub     *pubsub.Subscription
+	topic   *pubsub.Topic
+	parties party.IDSlice
+}
+
+func NewNetwork(parties party.IDSlice, sub *pubsub.Subscription, topic *pubsub.Topic) *Network {
+	c := &Network{
+		sub:     sub,
+		topic:   topic,
+		parties: parties,
+	}
+	return c
+}
+
+func (n *Network) Next(ctx context.Context) *protocol.Message {
+	msg, err := n.sub.Next(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	// Converting a pubsub.Message into a protocol.Message
```

Comment:
```suggestion
	// Unwrapping protocol message from PubSub message
```
Nit-pick to make it more readable. It is not quite transforming it, it is just fetching the protocol message marshalled data from the pubsub envelope.

Feel free to dissregard it if it doesn't make sense.